### PR TITLE
[ETCM-406] Distinguish between sync connecting and fully synced states

### PR DESCRIPTION
**Pull request by conqeror**
@kapke Do we need to distinguish between "sync connecting" and "sync completed" phase? Right now, I couldn't find a way to do so.

### CODE REVIEW

**Comment by kapke** [comment]
We need, unfortunately, I did think about it for a while and it seems that checking if last block number/hash changes within last ~30 seconds should be enough.
In that way we can start with offline/connecting state, then start polling both last block and sync status, and:
  - if sync returns syncing - it's pretty obvious
  - if sync returns false but last block did not stale - we're on top (I think we can move mining case out of equation, but it will be kind-of right in such case too)
  - if sync returns false but last block is the same as initial one or stale - we're offline/connecting

File: src/common/wallet-state.ts [hunk 1]
```diff
@@ -457,24 +471,47 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
...
+        ? previousSyncStatus.lastNewBlockTimestamp
+        : currentTimestamp
+
     if (syncing === false) {
-      const currentBlock = await web3.eth.getBlockNumber()
-      return {
-        mode: 'offline',
-        currentBlock: currentBlock,
+      if (
+        currentBlock === 0 ||
```
Comment (kapke) [inline src/common/wallet-state.ts:499]: Shouldn't initial block number be fetched when initializing wallet from scratch and that condition changed into `currentBlock === intialBlock`?

**Review by kapke — COMMENTED** [review]
One potential issue, besides that looks good to me!

**Review by qwhex — APPROVED** [review]
Let's get this merged after the last fix!

**Pull request merged**